src/ui/public/vis/vis.js

```diff
@@ -102,7 +103,7 @@ export default function VisFactory(Notifier, Private) {
   };
 
   Vis.prototype.clone = function () {
-    return new Vis(this.indexPattern, this.getState());
+    return new Vis(this.indexPattern, this.getState(), this.getUiState());
```

Comment:
`this.__uiStats` is an object, and `getUiState` simply returns it (by reference), which means the "clone" can change the parent value. You should be doing a `_.cloneDeep` somewhere to prevent this, probably in `getUiState` (so that other code reading that state can't change it either).